core/shared/src/main/scala/fs2/async/Promise.scala

```diff
@@ -0,0 +1,167 @@
+package fs2.async
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+import cats.implicits.{ catsSyntaxEither => _, _ }
+import cats.effect.{ Sync, Effect, IO }
+
+import fs2.Scheduler
+import fs2.internal.LinkedMap
+import java.util.concurrent.atomic.{AtomicBoolean,AtomicReference}
+
+import Promise._
+
+// TODO shall we move Effect and EC implicits back up here?
+// TODO scaladoc
+final class Promise[F[_], A] private [fs2] (ref: Ref[F, State[A]]) {
```

Comment:
Do `Ref` and `Promise` need to be parameterized by `F` or can that be on each method instead?